client/src/PagesBody/ContestsPage/ContestsPage.js

```diff
@@ -33,33 +36,47 @@ const ContestsPage = () => {
 
   useEffect(() => {
     async function fetchContestsData() {
-      const response = await fetch(
-        '/api/contests/card/data',
-        requestOptionsGET,
-      );
-      const result = await response.json();
+      let result = null;
+      if (
+        state.selectedRole === ROLE_NAME.PARTICIPANT &&
+        locationPath.state &&
+        locationPath.state.contestContent !== 'future'
+      ) {
+        result = await getContestsCards(ROLE_NAME.PARTICIPANT);
+      } else if (state.selectedRole === ROLE_NAME.MANAGER) {
+        result = await getContestsCards(ROLE_NAME.MANAGER);
+      } else if (state.selectedRole === ROLE_NAME.STAFF) {
+        result = await getContestsCards(ROLE_NAME.STAFF);
+      } else {
+        result = await getContestsCards();
+      }
+      if (Array.isArray(result)) {
+        result = removeNullsFromArray(result);
```

Comment:
What is the reason for this check?

Reply:
the reason is that we have a mess in mongoDB collections at the moment, some contests were removed by hand, but their ID still exists in another documents. utill the moment we have all endpoints and no one change data in MongoDB by hand, without that this side will become broken if there will be null in array

Reply:
Shuold probably make an issue for it, cleanup the db.

Reply:
issue is created!